src/components/shapes/defaults.js

```diff
@@ -116,4 +116,13 @@ function handleShapeDefaults(shapeIn, shapeOut, fullLayout) {
     if(noPath) {
         Lib.noneOrAll(shapeIn, shapeOut, ['x0', 'x1', 'y0', 'y1']);
     }
+
+    // Label options
+    coerce('label.text');
+    coerce('label.xanchor');
+    coerce('label.yanchor');
+    coerce('label.textangle', shapeType === 'line' ? 'auto' : 0);
+    coerce('label.position', '');
```

Comment:
Once you added `dflt: 'auto'` to `position` attribute, this line should become:
```suggestion
    coerce('label.position');
```